Added tip section for list components (#7609)
yamada-ui/yamada-ui: www/contents/components/(utilities)/for.mdx

<For />
```

:::tip

> If you want to display a single dataset as a list, use [List](/docs/components/list).

Reviewer: 🟡 risk: this makes `List` sound like a replacement for `For`, but they solve different parts of the problem. `For` still handles iterating over array data, while `List` provides the list structure. Could we phrase this as combining them instead, and make the same adjustment in the Japanese copy?

```suggestion
If you want to render array data as a list, combine [For](/docs/components/for) with [List](/docs/components/list).
```

Author: How do they render when combined?

Reviewer: Like this: `List` owns the list semantics/structure, and `For` maps the array into `List.Item`s.

```tsx
<List.Root>
  <For each={items}>
    {(item, index) => <List.Item key={index}>{item.children}</List.Item>}
  </For>
</List.Root>

Author: Thank you! Fixed.
